src/telemetry/telemetry.c

```diff
@@ -232,6 +237,246 @@ add_job_counts(JsonbParseState *state)
+		/* the jsonb object cannot be created in the SPI context or it will be lost */
+		spi_context = MemoryContextSwitchTo(old_context);
+		add_errors_by_sqlerrcode_internal(parse_state,
+										  TextDatumGetCString(record_jobtype),
+										  sqlerrs_jsonb);
+		old_context = MemoryContextSwitchTo(spi_context);
+	}
+
+	res = SPI_finish();
+
+	Assert(res == SPI_OK_FINISH);
+}
+
+static JsonbValue *
+add_job_stats_internal(JsonbParseState *state, const char *job_type, TelemetryJobStats *stats)
+{
+	JsonbValue key = {
+		.type = jbvString,
+		.val.string.val = pstrdup(job_type),
+		.val.string.len = strlen(job_type),
+	};
+	pushJsonbValue(&state, WJB_KEY, &key);
+	pushJsonbValue(&state, WJB_BEGIN_OBJECT, NULL);
+
+	ts_jsonb_add_int64(state, "total_runs", stats->total_runs);
+	ts_jsonb_add_int64(state, "total_successes", stats->total_successes);
+	ts_jsonb_add_int64(state, "total_failures", stats->total_failures);
+	ts_jsonb_add_int64(state, "total_crashes", stats->total_crashes);
+	ts_jsonb_add_int32(state, "max_consecutive_failures", stats->max_consecutive_failures);
+	ts_jsonb_add_int32(state, "max_consecutive_crashes", stats->max_consecutive_crashes);
+	ts_jsonb_add_interval(state, "total_duration", stats->total_duration);
+	ts_jsonb_add_interval(state, "total_duration_failures", stats->total_duration_failures);
+
+	return pushJsonbValue(&state, WJB_END_OBJECT, NULL);
+}
+
+static void
+add_job_stats_by_job_type(JsonbParseState *parse_state)
+{
+	StringInfo command;
+	int res;
+	MemoryContext old_context = CurrentMemoryContext, spi_context;
+	SPITupleTable *tuptable = NULL;
+
+	const char *command_string =
+		"SELECT ("
+		"	case WHEN q.proc_schema = \'_timescaledb_internal\' AND q.proc_name = \'policy_retention\' then q.proc_name::text "
+		"	WHEN q.proc_schema = \'_timescaledb_internal\' AND q.proc_name = \'policy_compression\' then q.proc_name::text "
+		"	WHEN q.proc_schema = \'_timescaledb_internal\' AND q.proc_name = \'policy_reorder\' then q.proc_name::text "
+		"	WHEN q.proc_schema = \'_timescaledb_internal\' AND q.proc_name = \'policy_refresh_continuous_aggregate\' then q.proc_name::text "
+		"	ELSE \'user_defined_action\'::text "
+		"	end"
+		") as job_type, sum(total_runs)::bigint as total_runs, sum(total_successes)::bigint as "
+		"total_successes,"
+		"sum(total_failures)::bigint as total_failures, sum(total_crashes)::bigint as "
+		"total_crashes,"
+		"sum(total_duration) as total_duration, sum(total_duration_failures) as total_duration_failures, max(consecutive_failures)::int as "
+		"max_consecutive_failures, max(consecutive_crashes)::int as max_consecutive_crashes "
+		"FROM "
+		"(select j.proc_schema, j.proc_name, s.total_runs, s.total_successes, s.total_failures,"
+		"s.total_crashes, s.total_duration, s.total_duration_failures, s.consecutive_crashes, s.consecutive_failures "
+		"FROM "
+		"_timescaledb_internal.bgw_job_stat s join _timescaledb_config.bgw_job j on j.id = "
+		"s.job_id) q "
+		"GROUP BY job_type";
```

Comment:
```suggestion
    "SELECT "
    "  CASE "
    "    WHEN j.proc_schema = \'_timescaledb_internal\' AND j.proc_name ~ \'^policy_(retention|compression|reorder|refresh_continuous_aggregate)$\' THEN j.proc_name::text  "
    "    ELSE \'user_defined_action\'::text  "
    "  END AS job_type, "
    "  sum(s.total_runs)::bigint AS total_runs, "
    "  sum(s.total_successes)::bigint AS total_successes, "
    "  sum(s.total_failures)::bigint AS total_failures, "
    "  sum(s.total_crashes)::bigint AS total_crashes, "
    "  sum(s.total_duration) AS total_duration, "
    "  sum(s.total_duration_failures) AS total_duration_failures, "
    "  max(s.consecutive_failures)::int AS max_consecutive_failures, "
    "  max(s.consecutive_crashes)::int AS max_consecutive_crashes  "
    "FROM "
    "  _timescaledb_internal.bgw_job_stat s "
    "  JOIN _timescaledb_config.bgw_job j ON j.id = s.job_id "
    "GROUP BY "
    "  job_type "
```
I didn't test but just for better code readability

Reply:
beautiful, thank you! Changed it as you suggest